### PR TITLE
Improvements impersonate

### PR DESCRIPTION
**Pull request by santib**

### CODE REVIEW

File: app/objects/impersonation/verifier.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        message_verifier.verified(_1, purpose: PURPOSE)
+      }
+    end
+
+    private
+
+    def message_verifier
+      @message_verifier ||= ActiveSupport::MessageVerifier.new(
+        Rails.application.secret_key_base,
+        digest: 'SHA256', serializer: JSON
```
Comment (santib) [inline app/objects/impersonation/verifier.rb:29]: in Rails 7.1 will be able to add `url_safe: true` option and remove the Base64 stuff.

File: app/objects/impersonation/verifier.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+module Impersonation
+  class Verifier
+    EXPIRATION = 5.minutes
+    PURPOSE = :impersonation
+
+    def sign!(data)
+      data.then {
+        message_verifier.generate(_1, expires_in: EXPIRATION, purpose: PURPOSE)
+      }.then {
+        Base64.urlsafe_encode64(_1)
```
Comment (mrodriguez90) [inline app/objects/impersonation/verifier.rb:12]: Is it possible to replace`_1` with `it`?

Comment (santib) [reply app/objects/impersonation/verifier.rb:12]: not yet, we need to wait til Christmas when Ruby 3.4 is going to be released 😅

Comment (santib) [reply app/objects/impersonation/verifier.rb]: I ended up removing the `then` chain now that we have only 2 steps

**Pull request merged**